internal/distro/rhel8/imagetype.go

```diff
@@ -346,31 +348,33 @@ func overridePackageNames(packages []string) []string {
 }
 
 // checkOptions checks the validity and compatibility of options and customizations for the image type.
-func (t *imageType) checkOptions(customizations *blueprint.Customizations, options distro.ImageOptions, containers []container.Spec) error {
+func (t *imageType) checkOptions(customizations *blueprint.Customizations, options distro.ImageOptions, containers []container.Spec) ([]string, error) {
 
+	// holds warnings (e.g. deprecation notices)
+	warnings := []string{}
```

Comment:
I suggest to define `warnings` as
```go
var warnings []string
```
and then just `append()` to it. This way, the function will return `nil` if there are not warnings, instead of an empty slice. So the behavior is now inconsistent with the same same functions defined in distros that do not return any warnings.